pinc/CharacterSelector.inc

```diff
@@ -1,241 +1,189 @@
+                    "lower" => [249, 250, 251, 252],
+                ],
+                [
+                    "code" => "+",
+                    "title" => _('More'),
+                    "upper" => [0x24, 0xa3, 0xa5, 0xa1, 0xa9, 0xab, 0x20, 0xdd, 0x20, 0xc7, 0xd0, 0xd1, 0xde, 0xa6, 0xaa, 0xba, 0xb4, 0xa8, 0xaf, 0xb8, 0xb0, 0xb9, 0xb2, 0xb3],
+                    "lower" => [0xa2, 0xa4, 0x20, 0xbf, 0xae, 0xbb, 0xdf, 0xfd, 0xff, 0xe7, 0xf0, 0xf1, 0xfe, 0xa7, 0xbc, 0xbd, 0xbe, 0xb6, 0xac, 0xb7, 0xb1, 0xd7, 0xf7, 0xb5],
+                ],
+            ];
+        }
     }
-    
-    private function echo_selector( $id, $title, $option_args )
+
+    function draw()
     {
-        global $utf8_site;
-    
-        echo '<td class="right-align">';
-        $title_safe = attr_safe($title);
-        echo <<<EOS
-    <select
-        name="tChars{$id}"
-        ID="tChars{$id}"
-        title="{$title_safe}"
-        onchange="if (this.options[selectedIndex].value !=0){top.insertCharacter(this.options[selectedIndex].value);}"
-        class="dropchars"
-    >
-EOS;
-    
-        foreach ( $option_args as $arg )
+        $selector_string = "<div id='selector_row'>";
+        $row_string = "";
+
+
+        foreach ($this->picker_set as $picker)
         {
-            if ( is_string($arg) )
-            {
-                echo "<option value='0'>{$arg}</option>\n";
-            }
-            elseif ( is_array($arg) )
+            $code = maybe_utf8_decode($picker["code"]);
+            $safe_code = bin2hex($code);
+            $title = maybe_utf8_decode($picker['title']);
+            $selector_string .= "<button type='button' id='_$safe_code' class='selector_button' title='$title'>" . html_safe($code) . "</button>";
+            $row_string .= "<div class='_$safe_code key-block'>\n";
+            $row_string .= $this->draw_row($picker["upper"]);
+            $row_string .= $this->draw_row($picker["lower"]);
+            $row_string .= "</div>\n";
+        }
+
+        $selector_string .= "</div>\n";
+
+        echo "<div class='char_box'><input type='text' readonly class='big_text' id='large_char'></div>\n";
+        echo "<div id='char-selector' class='nowrap'>$selector_string$row_string</div>\n";
+    }
+
+    function draw_row($chars)
+    {
+
+        $row = "<div class='table-row'>";
+
+        foreach($chars as $character)
+        {
+            $row .= "<div class='table-cell'>";
+            if (0x20 == $character)
             {
-                foreach ( $arg as $codepoint )
-                {
-                    if ( $codepoint <= 255 || $utf8_site )
-                    {
-                        echo "<option value='$codepoint'>&#{$codepoint};</option>\n";
-                    }
-                }
+                $row .= "<button type='button' class='picker invisible'></button>";
             }
             else
             {
-                assert( 0 );
+                $title_string = attr_safe(IntlChar::charName($character));
```

Comment:
Please update PHP section in `SETUP/installation.txt` to mention that the Intl PHP extension must be installed. On Ubuntu this is provided by the `php-intl` package.